plugins/nf-amazon/src/main/nextflow/cloud/aws/batch/AwsBatchTaskHandler.groovy

```diff
@@ -396,18 +396,18 @@ class AwsBatchTaskHandler extends TaskHandler implements BatchHandler<String,Job
     }
 
     protected String resolveJobDefinition0(String container) {
-        if( jobDefinitions.containsKey(container) )
-            return jobDefinitions[container]
+        final req = makeJobDefRequest(container)
+        final token = req.getParameters().get('nf-token')
```

Comment:
Does `nf-token` correspond to the process name?

Reply:
Nope, it's a unique hash determined by the request settings https://github.com/nextflow-io/nextflow/blob/master/plugins/nf-amazon/src/main/nextflow/cloud/aws/batch/AwsBatchTaskHandler.groovy#L437-L441

Reply:
Perfect